athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandler.java

```diff
@@ -251,8 +253,12 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
             LOGGER.info("MetadataHandler=GcsMetadataHandler|Method=doGetSplits|Message=StorageSplit JSON\n{}",
                     storageSplitJson);
             Split.Builder splitBuilder = Split.newBuilder(spillLocation, makeEncryptionKey())
-                    .add(CLASSIFICATION_GLUE_TABLE_PARAM, table.getParameters().get(CLASSIFICATION_GLUE_TABLE_PARAM))
+                    .add(FILE_FORMAT, table.getParameters().get(CLASSIFICATION_GLUE_TABLE_PARAM))
                     .add(STORAGE_SPLIT_JSON, storageSplitJson);
+
+            for (FieldVector fieldVector : partitions.getFieldVectors()) {
+                splitBuilder.add(fieldVector.getName(), fieldVector.getReader().readObject().toString());
+            }
```

Comment:
It looks you are putting a mapping of partition column name -> partition value as metadata onto the split, right?

However, it looks like you are just adding only the first partition value but its possible to have more than one partition value for each partition column, right?

So how does this work in that case?

Reply:
getPartitionsFolderLocationUri in src/main/java/com/amazonaws/athena/connectors/gcs/common/PartitionUtil.java uses this same pattern. Is it always the case that the partition has one value? If not, then we will want to revisit that logic as well.

Reply:
Actually Henry noticed that in PartitionUtil.java you're setting the reader position, so the logic in that class is fine

Reply:
The logic is correct. Earlier we relied on getPartitionsFolderLocationUri to setPosition() but set the explicit position in the same block of code now.